async-openai/src/types/assistant.rs

```diff
@@ -129,8 +129,11 @@ pub enum AssistantsApiResponseFormatOption {
 /// An object describing the expected output of the model. If `json_object` only `function` type `tools` are allowed to be passed to the Run. If `text` the model can return text or any value needed.
 #[derive(Clone, Serialize, Debug, Deserialize, PartialEq, Default)]
 pub struct AssistantsApiResponseFormat {
-    /// Must be one of `text` or `json_object`.
+    /// Must be one of `text`, `json_object` or `json_schema`.
     pub r#type: AssistantsApiResponseFormatType,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub json_schema: Option<serde_json::Value>,
```

Comment:
Would this be better as a struct? Like the functions api, the actual json schema is a field inside of an object with defined fields in the docs:
![2024-08-12-205225_389x547_scrot](https://github.com/user-attachments/assets/efe7d357-ebce-40ce-8efd-8cc77904e3a2)